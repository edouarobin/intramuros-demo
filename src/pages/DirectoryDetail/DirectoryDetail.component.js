// @flow

import React, { PureComponent } from 'react';
import {
  FlatList,
  ScrollView,
  StyleSheet,
  Text,
  View,
  StatusBar,
} from 'react-native';

import { ContactCard, GoBackButton } from 'Intramuros/src/components';
import globalStyle, {
  BIG_SCREEN_PADDING_OFFSET,
} from 'Intramuros/src/style/globalStyle';

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    flex: 1,
    paddingHorizontal: BIG_SCREEN_PADDING_OFFSET,
  },
  titleContainer: {
    alignItems: 'center',
    paddingBottom: 10,
  },
  title: {
    fontFamily: 'Lato-Bold',
    fontSize: 20,
    // fontWeight: 'bold',
    color: 'black',
  },
  separator: {
    backgroundColor: globalStyle.colors.yellow,
    height: 2,
    width: '30%',
    alignSelf: 'center',
  },
});

declare type PropsType = {
  navigation: NavigationPropsType<{
    directory: DirectoryType,
  }>,
};

export default class DirectoryDetail extends PureComponent<PropsType> {
  render() {
    const directory = this.props.navigation.state.params.directory;
    return (
      <ScrollView
        style={{ backgroundColor: 'white' }}
        showsVerticalScrollIndicator={false}
      >
        <GoBackButton
          navigation={this.props.navigation}
          color={globalStyle.colors.mainBlue}
        />
        <StatusBar
          barStyle="dark-content"
          translucent
          backgroundColor="transparent"
          animated={true}
        />
        <View style={styles.container}>
          <View style={styles.titleContainer}>
            <Text style={styles.title}>{directory.title}</Text>
          </View>
          <View style={styles.separator} />
          <FlatList
            data={directory.contacts}
            renderItem={({ item }) => <ContactCard contact={item} />}
          />
        </View>
      </ScrollView>
    );
  }
}
