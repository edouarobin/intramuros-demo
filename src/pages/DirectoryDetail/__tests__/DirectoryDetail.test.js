// @flow-weak

import React from 'react';
import renderer from 'react-test-renderer';

import DirectoryDetail from '../DirectoryDetail.component';

it('Fully renders the DirectoryDetails page', () => {
  const mockDirectory = {
    title: 'Mairie',
  };
  const tree = renderer
    .create(
      <DirectoryDetail
        navigation={{
          state: {
            params: { directory: mockDirectory },
          },
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
