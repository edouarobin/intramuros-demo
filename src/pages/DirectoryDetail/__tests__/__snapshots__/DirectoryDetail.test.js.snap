// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fully renders the DirectoryDetails page 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Mairie
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFD153",
            "height": 2,
            "width": "30%",
          }
        }
      />
      <RCTScrollView
        data={undefined}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onViewableItemsChanged={undefined}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</RCTScrollView>
`;
